### DISCUSSION TITLE
Netvox Z311A support

### DISCUSSION THREAD

**Discussion by pablopoo** [zigpy/zha-device-handlers#2393] [Q&A]
Hi, I'm trying to connect a netvox Z311A contact sensor. So far the battery level is working but the contact sensor not.
looking on the Zone Status Change Notification Payload on the device documentation I see that is not following the standard:

This is the notification payload standard:
![image](https://github.com/zigpy/zha-device-handlers/assets/13492043/a12dff92-8f1d-4eda-8819-7600d08e4226)

and this is the device payload:
![image](https://github.com/zigpy/zha-device-handlers/assets/13492043/5cd7305d-6d9d-4e54-8e28-d07a14d34e93)

For what I understand, the device is not sending the zone Id and delay as separated payload parameters, and instead send the zone id inside the extended status parameter.

Should I do something like this?

`    manufacturer_client_commands = {
        0x0000: (
            "status_change_notification",
            (
                IasZone.ZoneStatus,
                t.bitmap8,
                t.Optional(t.uint8_t),
                t.Optional(t.uint16_t),
            ),
            False,
        )
    }`

I'm new to this, any hints will be appreciated. Thanks!

**Comment by Hedda** [comment]
@pablopoo Start by opening a new issue with "Device support request" -> https://github.com/zigpy/zha-device-handlers/issues

There you provide the requested device signature, diagnostic information, logs, etc. which makes it easier for others to answer and help.

PS: This is in ZHA docs -> https://www.home-assistant.io/integrations/zha#how-to-add-support-for-new-and-unsupported-devices

**Reply by pablopoo** [reply]
Thanks for the tip. I did implemented the previous solution, defining the custom command mapping, and then it starting recognize the zone status correctly 

`manufacturer_client_commands = { 0x0000: ( "status_change_notification", ( IasZone.ZoneStatus, t.bitmap8, ), False, ) } `

The only thing I have pending is reading the 8 bits extended status message.

I will upload the full code to a repo. I saw that other netvox devices follow the same message structure pattern.
